include/restapi/search_wrapper.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <stdexcept>
+
+#include <index.h>
+#include <pq_flash_index.h>
+
+namespace diskann {
+  class SearchResult {
```

Comment:
Maybe we could templatize this as <T, TagT>? Are you avoiding do that because of serialization concerns?